### PR TITLE
Copy packages and download packages

### PR DESCRIPTION
**Pull request by lasote**
Base for kind or binary PR

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -292,9 +302,38 @@ def remove(self, *args):
...
-        self._manager.remove(args.pattern, package_ids_filter=args.packages, build_ids=args.builds,
+        self._manager.remove(args.pattern, package_ids_filter=args.packages,
+                             build_ids=args.builds,
                              src=args.src, force=args.force, remote=args.remote)
 
+    def copy(self, *args):
+        """ Copy packages to another user/channel
+        """
+        parser = argparse.ArgumentParser(description=self.copy.__doc__, prog="conan copy")
+        parser.add_argument("reference", nargs='?', default="",
```
Comment (memsharded) [inline conans/client/command.py]: check nargs = "?" seems wrong

File: conans/client/command.py [hunk 2]
```diff
@@ -292,9 +302,38 @@ def remove(self, *args):
...
+                            help='reference'
+                            'e.g., OpenSSL/1.0.2e@lasote/stable')
+        parser.add_argument("user_channel", nargs='?', default="",
```
Comment (memsharded) [inline conans/client/command.py]: same here:
nargs = "?"

File: conans/client/manager.py [hunk 3]
```diff
@@ -256,6 +275,22 @@ def file_manager(self):
...
+        @param package_ids: list of ids or [] for all list
+        @param username: Destination username
+        @param channel: Destination channel
+        @param remote: install only from that remote
+        """
+        copier = PackageCopier(self._paths, self._user_io)
+
+        if package_ids:
+            copier.copy(reference, package_ids, username, channel, force)
+        else:  # Not specified packages, download all
```
Comment (memsharded) [inline conans/client/manager.py]: download all?
I'd prefer here a single copier.copy() call and a:
```python
package_ids = package_ids or os.listdir(self._paths.packages(reference))
```

File: conans/test/integration/copy_packages_test.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+from conans.test.utils.cpp_test_files import cpp_hello_conan_files
+from conans.paths import CONANFILE
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.util.files import rmdir
+
+
+class CopyPackagesTest(unittest.TestCase):
+
+    def test_copy_command(self):
+        test_server = TestServer([("*/*@*/*", "*")],  # read permissions
```
Comment (memsharded) [inline conans/test/integration/copy_packages_test.py]: server not needed at all?

File: conans/test/integration/copy_packages_test.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                                 [],  # write permissions
+                                 users={"lasote": "mypass"})  # exported users and passwords
+        self.servers = {"default": test_server}
+        self.client = TestClient(servers=self.servers,  users=[("lasote", "mypass")])
```
Comment (memsharded) [inline conans/test/integration/copy_packages_test.py]: Too much "self"

File: conans/test/package_copier_test.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        self._create_package(reference, "0101001", paths)
+        self._create_package(reference, "2222222", paths)
+
+        # Copy all to destination
+        copier.copy(reference, ["0101001", "2222222"], "lasote", "stable", force=False)
+        new_reference = ConanFileReference.loads("Hello/0.1@lasote/stable")
+        self._assert_conanfile_exists(new_reference, paths)
+        self._assert_package_exists(new_reference, "0101001", paths)
+        self._assert_package_exists(new_reference, "2222222", paths)
+        self.assertIn("Copied Hello/0.1@lasote/testing to Hello/0.1@lasote/stable", output)
```
Comment (memsharded) [inline conans/test/package_copier_test.py]: I suggest reviewing this output once the scoped output PR is merged, I'd say this output can be improved:
- Copied all
- Hello/0.1@lasote/testing:ID copied to Hello/0.1@user/stable

Comment (lasote) [inline conans/test/package_copier_test.py]: GRRR! I was first XDD

**Comment by memsharded** [comment]
Ok, minor details, can be merged after solving them.

File: conans/test/package_copier_test.py (hunk 6 above)
Comment (memsharded) [inline conans/test/package_copier_test.py]: I mean, merge it as-is.
Then merge scoped-output
Then review output with scopes, to check for consistency of messages format.

**Pull request merged**